src/components/VenueList/VenueCard/ImageSlider/hooks/useSliderControls.js

```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect } from 'react';
+
+export const useSliderControls = (images) => {
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const handlePrevious = (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsLoading(true);
+    setCurrentIndex((previousIndex) =>
+      previousIndex === 0 ? images.length - 1 : previousIndex - 1,
+    );
+  };
+
+  const handleNext = (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsLoading(true);
+    setCurrentIndex((previousIndex) =>
+      previousIndex === images.length - 1 ? 0 : previousIndex + 1,
+    );
+  };
+
+  const handleImageLoad = () => {
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
```

Comment:
Here, you set all loaders to true every time the images change. This is probably why there is a bug where sometimes all images have loading indicators. Don't use `useEffect` like that here.

<img width="954" height="935" alt="image" src="https://github.com/user-attachments/assets/f48f49d5-502a-4f3e-9d1d-202a18d01530" />